From a25b2b9ae9554cd7a25c7be8a3c1a00fb628c456 Mon Sep 17 00:00:00 2001
From: Matt Whitlock <pypkgconf@mattwhitlock.name>
Date: Sun, 2 Jun 2024 12:29:13 -0400
Subject: [PATCH] meson.build: use system-installed libpkgconf if found

The Meson manual [advises](https://mesonbuild.com/Subprojects):

> When building distro packages it is very important that you do not
> embed any sources. Some distros have a rule forbidding embedded
> dependencies so your project must be buildable without them or
> otherwise the packager will hate you.

So, first call `dependency()` to try to find a system-installed
`libpkgconf` before falling back to the WrapDB package.
---
 meson.build | 23 ++++++++++++++++++-----
 1 file changed, 18 insertions(+), 5 deletions(-)

diff --git a/meson.build b/meson.build
index fa63b9d..1bbd6ee 100644
--- a/meson.build
+++ b/meson.build
@@ -11,12 +11,25 @@ fs = import('fs')
 pymod = import('python')
 py = pymod.find_installation(modules: ['cffi'], pure: false)
 
-pkgconf = subproject('pkgconf')
-libpkgconf = pkgconf.get_variable('dep_libpkgconf')
+libpkgconf = dependency('libpkgconf', required: false)
+if libpkgconf.found()
+    foreach cflag : run_command('pkg-config', '--cflags-only-I', libpkgconf.name(), check: true).stdout().split()
+        if cflag.startswith('-I') and fs.is_file(cflag.substring(2) / 'libpkgconf' / 'libpkgconf.h')
+            libpkgconf_include = cflag.substring(2)
+            break
+        endif
+    endforeach
+    cdata = configuration_data()
+    cdata.set_quoted('PKG_DEFAULT_PATH', run_command('pkg-config', '--variable=pc_path', 'pkg-config', check: true).stdout())
+    build_static = '-DPKGCONFIG_IS_NOT_STATIC'
+else
+    pkgconf = subproject('pkgconf')
+    libpkgconf = pkgconf.get_variable('dep_libpkgconf')
+    libpkgconf_include = meson.global_source_root() / 'subprojects' / 'pkgconf-2.1.1'
+    cdata = pkgconf.get_variable('cdata')
+    build_static = pkgconf.get_variable('build_static')
+endif
 libpkgconf_c = '_libpkgconf.c'
-libpkgconf_include = meson.global_source_root() / 'subprojects' / 'pkgconf-2.1.1'
-cdata = pkgconf.get_variable('cdata')
-build_static = pkgconf.get_variable('build_static')
 
 c_bindings = custom_target(
     'libpkgconf_c',
-- 
2.45.1

